proto-lens-protoc/app/Data/ProtoLens/Compiler/Generate.hs

```diff
@@ -904,6 +908,8 @@ messageInstance :: Env RdrNameStr -> T.Text -> MessageInfo OccNameStr -> [RawIns
 messageInstance env protoName m =
     [ funBind "messageName" $ match [wildP] $
           var "Data.Text.pack" @@ string (T.unpack protoName)
+    , funBind "messageDescriptor" $ match [wildP] $
+          var "Data.ByteString.pack" @@ list msgDescriptor
```

Comment:
I think it would be better to use OverloadedStrings (which this file already supports), e.g.

```
messageDescriptor _ = "abcde"
```

The benefits would be: (1) the syntax is more compact, and (2) under the hood, GHC optimizes away the intermediate string/list and stores the packed bytestring directly in the generated code.

I think it would be something like:

```
funBind "messageDescriptor" $ match [wildP] $ string msgDescriptor
...
  where
    msgDescriptor = fmap (toEnum . fromEnum) . BS.unpack . buildMessage $
       messageDescriptor m
```

(which avoids needing to add an import of Data.ByteString.Char in the generated code).

Reply:
OverloadedStrings wasn't active in the generated protos.  I was trying to follow along with  the `messageName` thing which is using `Data.Text.unpack`.  I updated this to use a string, though.